tests/functional/object/mcg/ui/test_mcg_ui.py

```diff
@@ -441,3 +442,207 @@ def test_bucket_create(self, setup_ui_class_factory):
+            delete_via="three_dots", expect_fail=False, resource_name=bucket_to_delete
+        )
+
+        bucket_ui.nav_object_storage_page()
+        bucket_ui.nav_buckets_page()
+
+        updated_buckets = bucket_ui.get_buckets_list()
+        assert (
+            bucket_to_delete not in updated_buckets
+        ), f"Bucket {bucket_to_delete} was not deleted successfully"
+        logger.info(f"Successfully deleted bucket: {bucket_to_delete}")
+
+    def test_bucket_list_comparison(self, setup_ui_class_factory, mcg_obj):
+        """
+        Test that the bucket list from UI matches the bucket list from CLI.
+
+        Args:
+            setup_ui_class_factory (any): UI setup fixture.
+            mcg_obj (any): MCG CLI client fixture assumed to have a method list_buckets() returning list.
+
+        Returns:
+            None
+
+        Raises:
+            AssertionError: If UI bucket count doesn't match CLI bucket count.
+        """
+        setup_ui_class_factory()
+        bucket_ui = BucketsTab()
+        bucket_ui.nav_object_storage_page()
+        bucket_ui.nav_buckets_page()
+
+        # Get CLI bucket list for comparison
+        cli_buckets = list(mcg_obj.cli_list_all_buckets())
+        logger.info(f"CLI bucket count: {len(cli_buckets)}")
+
+        all_ui_buckets = []
+
+        first_page_buckets = bucket_ui.get_buckets_list()
+        all_ui_buckets.extend(first_page_buckets)
+        logger.info(f"First page bucket count: {len(first_page_buckets)}")
+
+        if bucket_ui.has_pagination_controls() and len(first_page_buckets) == 100:
+            logger.info("Pagination detected, collecting buckets from additional pages")
+
+            while bucket_ui.navigate_to_next_page():
+                next_page_buckets = bucket_ui.get_buckets_list()
+                logger.info(f"Additional page bucket count: {len(next_page_buckets)}")
+                all_ui_buckets.extend(next_page_buckets)
+
+                if len(all_ui_buckets) >= len(cli_buckets):
+                    break
+
+        logger.info(f"Total UI bucket count across all pages: {len(all_ui_buckets)}")
+        logger.info(f"CLI bucket count: {len(cli_buckets)}")
+
+        ui_bucket_set = set(all_ui_buckets)
+        cli_bucket_set = set(cli_buckets)
+
+        missing_in_ui = cli_bucket_set - ui_bucket_set
+        if missing_in_ui:
+            logger.warning(f"Buckets in CLI but missing in UI: {missing_in_ui}")
+
+        missing_in_cli = ui_bucket_set - cli_bucket_set
+        if missing_in_cli:
+            logger.warning(f"Buckets in UI but missing in CLI: {missing_in_cli}")
+
+        assert (
+            len(all_ui_buckets) >= 2
+        ), "Expected at least 2 buckets (OBC and S3) but found less"
+
+        assert len(all_ui_buckets) == len(
+            cli_buckets
+        ), f"UI bucket count ({len(all_ui_buckets)}) does not match CLI bucket count ({len(cli_buckets)})"
+
+    def test_bucket_pagination(self, setup_ui_class_factory, mcg_obj):
```

Comment:
If this test takes too long, you might want to make it a tier2 or tier3 instead of a tier1

Reply:
yes, test test_bucket_pagination takes 29 min due to [verification run](https://jenkins-csb-odf-qe-ocs4.dno.corp.redhat.com/job/qe-trigger-test-pr/8438/testReport/tests.functional.object.mcg.ui.test_mcg_ui/TestBucketCreate/ )
makes sense to me too